### ISSUE TITLE
Mirador/OpenSeaDragon unable to display images served by the rosa-iiif layer for FSI

### ISSUE THREAD

**Issue by najamansari** [jhu-digital-manuscripts/rosa2#106]
Hello,

We are using the rosa-iiif image endpoint to serve our images hosted in FSI. The IIIF 1.1 version of the endpoint was working perfectly but the IIIF 2.0 version provided by rosa2 cannot be displayed by Mirador and even the base OpenSeaDragon viewer.

Let me know if I can provide any information to help you. Thanks.

**Comment by markpatton** [comment]
Najam,

Try directly hitting the endpoint with a command line tool like curl, or
even your web browser and see what happens. If it doesn't work as you
expect, take a look in the log of your application server and see what it
says. It may be a configuration issue. Check for example if
iiif-servlet.properties is pointing at your image server.

If you have more trouble, please tell us what your configuration is and
what error messages are in your log.

Mark

**Comment by najamansari** [comment]
Hi Mark,

The API is accessible both through Curl and by accessing from the browser.

When I used the API with Mirador, Mirador was able to get both the info JSON and the thumbnails for the images. However it could not display the actual images.

I then tried using an image directly with OpenSeaDragon (the IIIF client used internally by Mirador). OSD sent out an info.json request but did not load the image (or send out any requests for that). The IIIF 1.x version of the rosa API is working pefectly with both OSD and Mirador which makes me think that perhaps there is some missing data in the info JSON which OSD needs in order to display the images correctly.

Perhaps I should get in touch with the OSD team to see what missing data (if any) we need in the info JSON?

**Comment by markpatton** [comment]
Najam,

I can only say that we are using Mirador and its working pretty well. Are
there any error messages in your log on the application server? Can you
point us at your iiiif endpoints (image and presentation) so we can test
them?

Mark

**Comment by najamansari** [comment]
Hi Mark,

My apologies for the delayed response.

You may view the manifest containing the IIIF 2.0 URLs at: http://dev.acit.com/apps/pcpa/manifests/iiif2.0.json

The same manifest with IIIF 1.0 URLs may be viewed at: http://mirador.artstor.acit.com:9867/iiif/ssc/collection_id/7729716/manifest

Thanks.

**Comment by jabrah** [comment]
We are getting the same results here. In Mirador, the thumbnails display correctly, but when navigating to see the full page, nothing is displayed. I did the same as you and opened the image with OpenSeadragon directly with similar results. 

One thing I did try, however, was putting the _info.json_ for the image directly into OSD to pay around. My initial impression is that it does not like the current format of the `tiles` with the width set to -1. Either eliminating the tiles object from _info.json_ or setting a width > 0 allowed OSD to display the image. I suspect this would fix the behavior in Mirador as well, but have not been able to test this.

At the moment it is hard to say if something is configured incorrectly with the IIIF image endpoint, or the FSI image server, or if there is a bug with the image endpoint code. Will look into it when I get some time.

(I did not see any errors coming from the IIIF image endpoint, which would appear in the server logs. If you do see errors here, it would be helpful to know what the errors are.)

**Comment by najamansari** [comment]
Hello,

I went through the source of rosa for IIIF 1.1 and noticed that the tile widths and heights were being set to 1000 in there. I replicated the same in the rosa2 code. Additionally, I changed the scaleFactors to [1, 2, 4] (based on what I've seen in info.jsons from the IIP image server). With these changes, I recompiled and redeployed the package and now the images are showing up in Mirador.
